rules/subrepo_rules.build_defs

```diff
@@ -126,7 +139,7 @@ def new_http_archive(name:str, urls:list, build_file:str=None, build_file_conten
             'build': [build_file],
         },
         tools = [CONFIG.JARCAT_TOOL],
-        outs = [name],
+        outs = [f"{name}/{path}" for path in files] if files else [name],
```

Comment:
I'm not sure about this part, as the out of this rule becomes the root for the subrepo rule below.

If I specify `files = ["path/to/file"]`, then `path/to/file` is considered the root for the subrepo. I.e. https://github.com/thought-machine/please/blob/master/src/parse/asp/builtins.go#L1043 and https://github.com/thought-machine/please/blob/master/src/core/subrepo.go#L57-L58

I think ideally the output would be the single directory with only the files that are specified.